plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt

```diff
@@ -739,10 +741,10 @@ private fun PackageJson.getScopeDependencies(type: YarnDependencyType) =
 
 private fun getYarnExecutable(workingDir: File): File {
     val yarnrcFile = workingDir.resolve(YARN2_RESOURCE_FILE)
-    val yarnConfig = yamlMapper.readTree(yarnrcFile)
-    val yarnPath = requireNotNull(yarnConfig[YARN_PATH_PROPERTY_NAME].textValueOrEmpty().ifBlank { null }) {
-        "No Yarn 2+ executable could be found in '$YARN2_RESOURCE_FILE'."
-    }
+    val yarnConfig = Yaml.default.parseToYamlNode(yarnrcFile.readText()).yamlMap
+    val yarnPath = yarnConfig.get<YamlScalar>(YARN_PATH_PROPERTY_NAME)?.content.orEmpty().ifBlank { null }
+
+    requireNotNull(yarnPath) { "No Yarn 2+ executable could be found in '$YARN2_RESOURCE_FILE'." }
```

Comment:
While at it, maybe simplify to

    val yarnPath = yarnConfig.get<YamlScalar>(YARN_PATH_PROPERTY_NAME)?.content

    require(!yarnPath.isNullOrEmpty()) { "No Yarn 2+ executable could be found in '$YARN2_RESOURCE_FILE'." }

Again, I'm unsure whether we need to cover the pathological bank case.